src/controllers/user-order.controller.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/example-shopping
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {repository, Filter, Where} from '@loopback/repository';
+import {UserRepository} from '../repositories';
+import {
+  post,
+  get,
+  patch,
+  del,
+  param,
+  requestBody,
+  HttpErrors,
+} from '@loopback/rest';
+import {Order} from '../models';
+
+/**
+ * Controller for User's Orders
+ */
+export class UserOrderController {
+  constructor(@repository(UserRepository) protected userRepo: UserRepository) {}
+
+  /**
+   * Create or update the orders for a given user
+   * @param userId User id
+   * @param cart Shopping cart
+   */
+  @post('/users/{userId}/orders', {
+    responses: {
+      '200': {
+        description: 'User.Order model instance',
+        content: {'application/json': {'x-ts-type': Order}},
+      },
+    },
+  })
+  async create(
+    @param.path.string('userId') userId: string,
+    @requestBody() order: Order,
+  ): Promise<Order> {
+    if (userId !== order.userId) {
+      throw new HttpErrors.BadRequest(
+        `User id does not match: ${userId} !== ${order.userId}`,
+      );
+    }
+    delete order.userId;
+    return await this.userRepo.orders(userId).create(order);
+  }
+
+  @get('/users/{userId}/orders', {
+    responses: {
+      '200': {
+        description: "Array of User's Orders",
+        content: {
+          'application/json': {
+            schema: {type: 'array', items: {'x-ts-type': Order}},
+          },
+        },
+      },
+    },
+  })
+  async find(
+    @param.path.string('userId') userId: string,
+    @param.query.string('filter') filter?: Filter,
+  ): Promise<Order[]> {
+    const orders = await this.userRepo
+      .orders(userId)
+      .find(filter, {strictObjectIDCoercion: true});
+    return orders;
+  }
+
+  @patch('/users/{userId}/orders', {
+    responses: {
+      '200': {
+        description: 'User.Order PATCH success count',
+        content: {'application/json': {'x-ts-type': Number}},
+      },
+    },
+  })
+  async patch(
```

Comment:
I would like to see a name meaningful for business logic, such as `updateOrders`.

Reply:
`update` does not make it clear whether it's a full replace or a partial patch. Let's use `patchOrders` please.